test/PINO_ode_tests.jl

```diff
@@ -0,0 +1,304 @@
+
+@testsetup module PINOODETestSetup
+using Lux, NeuralOperators
+
+function get_trainset(chain::DeepONet, bounds, number_of_parameters, tspan, dt)
+    p_ = [range(start = b[1], length = number_of_parameters, stop = b[2]) for b in bounds]
+    p = vcat([collect(reshape(p_i, 1, size(p_i, 1))) for p_i in p_]...)
+    t_ = collect(tspan[1]:dt:tspan[2])
+    t = reshape(t_, 1, size(t_, 1), 1)
+    (p, t)
+end
+
+function get_trainset(chain::Lux.Chain, bounds, number_of_parameters, tspan, dt)
+    tspan_ = tspan[1]:dt:tspan[2]
+    pspan = [range(start = b[1], length = number_of_parameters, stop = b[2])
+             for b in bounds]
+    x_ = hcat(vec(map(
+        points -> collect(points), Iterators.product([pspan..., tspan_]...)))...)
+    x = reshape(x_, size(bounds, 1) + 1, prod(size.(pspan, 1)), size(tspan_, 1))
+    p, t = x[1:(end - 1), :, :], x[[end], :, :]
+    (p, t)
+end
+export get_trainset
+end
+#Test Chain
+@testitem "Example Chain du = cos(p * t)" tags=[:pinoode] setup=[PINOODETestSetup] begin
+    using NeuralPDE, Lux, OptimizationOptimisers, NeuralOperators, Random
+    equation = (u, p, t) -> cos(p * t)
+    tspan = (0.0f0, 1.0f0)
+    u0 = 1.0
+    prob = ODEProblem(equation, u0, tspan)
+    chain = Chain(
+        Dense(2 => 10, Lux.tanh_fast), Dense(10 => 10, Lux.tanh_fast), Dense(10 => 1))
+    x = rand(2, 50, 10)
+    θ, st = Lux.setup(Random.default_rng(), chain)
+    b = chain(x, θ, st)[1]
+
+    bounds = [(pi, 2pi)]
+    number_of_parameters = 300
+    strategy = StochasticTraining(300)
+    opt = OptimizationOptimisers.Adam(0.01)
+    alg = PINOODE(chain, opt, bounds, number_of_parameters; strategy = strategy)
+    sol = solve(prob, alg, verbose = false, maxiters = 5000)
+    ground_analytic = (u0, p, t) -> u0 + sin(p * t) / (p)
+    dt = 0.025f0
+    p, t = get_trainset(chain, bounds, number_of_parameters, tspan, dt)
+    ground_solution = ground_analytic.(u0, p, t)
+    predict_sol = sol(reduce(vcat, (p, t)))
+    @test ground_solution≈predict_sol rtol=0.05
+    p, t = get_trainset(chain, bounds, 100, tspan, 0.01)
+    ground_solution = ground_analytic.(u0, p, t)
+    predict_sol = sol(reduce(vcat, (p, t)))
```

Comment:
it doesn't match the documented interface though, so it's really odd. `sol.original(...)` would be allowed to do this, but what I was saying is that we shouldn't have an interface break here.

Reply:
How do you think it would be better ?

Reply:
`sol(t)` should do the normal interoplation w.r.t. `p` in the ODEProblem. You can overload `sol.interp` to do extra things and document that as well, but `sol(t)` is described as having very specific behavior which should be kept the same with all other ODE solvers.